File: HomeForYou/client/src/schemas/propertySchema.ts
import z from "zod";

const propertyCreationSchema = z.object({
  description: z.string(),
  price: z.number(),
  area: z.number(),
  neighborhood_id: z.number(),
  user_id: z.number(),
});

export type PropertyCreationSchema = z.infer<typeof propertyCreationSchema>;

const propertySchema = z.object({
  id: z.number(),
  description: z.string(),
  price: z.number(),
  area: z.number(),
  neighborhood_id: z.number(),
  user_id: z.number(),
});

export type PropertySchema = z.infer<typeof propertySchema>;

File: HomeForYou/server/src/schemas/propertySchema.py
from pydantic import BaseModel


class PropertyBase(BaseModel):
    description: str
    price: int
    area: int
    neighborhood_id: int


class PropertyCreate(PropertyBase):
    user_id: int


class PropertyRead(PropertyBase):
    id: int
    user_id: int

    class Config:
        orm_mode = True
